src/components/fields/derived.js

```diff
@@ -676,3 +677,15 @@ export const HoverColor = connectToContainer(UnconnectedColorPicker, {
     return plotProps;
   },
 });
+
+export const BinSize = connectToContainer(UnconnectedNumeric, {
+  modifyPlotProps: (props, context, plotProps) => {
+    const {localize: _} = context;
+    if (typeof plotProps.fullValue === 'string' && plotProps.fullValue[0] === 'M') {
+      plotProps.fullValue = plotProps.fullValue.substring(1);
+      plotProps.min = 1;
+      plotProps.max = 12;
+      plotProps.units = parseInt(plotProps.fullValue, 10) === 1 ? _('Month') : _('Months');
```

Comment:
does plotly.js accept any other format than M for months? D for days? seconds? I'd be surprised if it was only months :)

Reply:
Well actually, from the docs, it does look that months is the only option: 
```
If the axis `type` is "date", then you must convert the time to milliseconds. For example, to set the interval between ticks to one day, set `dtick` to 86400000.0. "date" also has special values "M<n>" gives ticks spaced by a number of months. `n` must be a positive integer. To set ticks on the 15th of every third month, set `tick0` to "2000-01-15" and `dtick` to "M3". To set ticks every 4 years, set `dtick` to "M48"
```

Either that or time in milliseconds, so I could add a richer widgets that allows to do either milliseconds or months